examples/social/src/lib.rs

```diff
@@ -251,33 +250,56 @@ pub struct Key {
 
 // Serialize keys so that the lexicographic order of the serialized keys corresponds to reverse
 // chronological order, then sorted by author, then by descending index.
-impl CustomSerialize for Key {
-    fn to_custom_bytes(&self) -> Result<Vec<u8>, ViewError> {
+impl Serialize for Key {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         let data = (
             (!self.timestamp.micros()).to_be_bytes(),
             &self.author,
             (!self.index).to_be_bytes(),
         );
-        Ok(bcs::to_bytes(&data)?)
+        let key = bcs::to_bytes(&data).unwrap();
+        serializer.serialize_bytes(&key)
```

Comment:
Aren't we already using BCS anyway?
So it shouldn't be necessary to explicitly use it here:
```suggestion
        data.serialize(serializer)
```